src/renderer/display_list.rs

```diff
@@ -1187,16 +1187,24 @@ fn fold_prefix_suffix(mut snippet: snippet::Snippet<'_>) -> snippet::Snippet<'_>
         snippet.source = &snippet.source[new_start..];
 
         for ann in &mut snippet.annotations {
-            let range_start = ann.range.start - new_start;
-            let range_end = ann.range.end - new_start;
-            ann.range = range_start..range_end;
+            let range_start = match ann.range.0 {
+                Bound::Unbounded => Bound::Unbounded,
+                Bound::Excluded(e) => Bound::Excluded(e - new_start),
+                Bound::Included(e) => Bound::Included(e - new_start),
+            };
+            let range_end = match ann.range.1 {
+                Bound::Unbounded => Bound::Unbounded,
```

Comment:
Hm, I'm wondering if this needs to be `Bound::Exclusive(ann.exclusive_end(foo) - new_start)` instead.